x-pack/plugins/cases/public/components/case_view/components/case_view_alerts.tsx

```diff
@@ -22,34 +24,48 @@ interface CaseViewAlertsProps {
 export const CaseViewAlerts = ({ caseData }: CaseViewAlertsProps) => {
   const { triggersActionsUi } = useKibana().services;
 
+  const alertIds = getManualAlertIds(caseData.comments);
   const alertIdsQuery = useMemo(
     () => ({
       ids: {
-        values: getManualAlertIds(caseData.comments),
+        values: alertIds,
       },
     }),
-    [caseData.comments]
+    [alertIds]
   );
 
-  const alertRegistrationContexts = useMemo(
-    () => getRegistrationContextFromAlerts(caseData.comments),
-    [caseData.comments]
+  const { isLoading: isLoadingAlertFeatureIds, data: alertFeatureIds } = useGetFeatureIds(
+    alertIds,
+    alertIdsQuery,
+    caseData.owner !== SECURITY_SOLUTION_OWNER
   );
 
-  const { isLoading: isLoadingAlertFeatureIds, data: alertFeatureIds } =
-    useGetFeatureIds(alertRegistrationContexts);
-
-  const configId =
-    caseData.owner === SECURITY_SOLUTION_OWNER ? `${caseData.owner}-case` : caseData.owner;
+  let configId = caseData.owner;
+  if (caseData.owner === SECURITY_SOLUTION_OWNER) {
+    configId = `${caseData.owner}-case`;
+  } else if (caseData.owner === GENERAL_CASES_OWNER && alertFeatureIds.includes('ml')) {
+    configId = `mlAlerts`;
```

Comment:
I presume it's not possible to import [ML_ALERTS_CONFIG_ID](https://github.com/elastic/kibana/blob/875268d55888b598bb765bd7d4adc3527d671de2/x-pack/plugins/ml/public/alerting/anomaly_detection_alerts_table/register_alerts_table_configuration.tsx#L149) due to circular dependency, but maybe we should defined it in the package or in the alerting plugin

Reply:
Also this `else if` looks like a hack to me, should we consider a generic way of resolving a config ID?

Reply:
Good idea!

Reply:
About making it more generic I have this issue here https://github.com/elastic/kibana/issues/143201 to do it at some point in the future (tech dept).

Reply:
As demo'ed to @XavierM , for a case with ML alerts, the alerts tab inside the Case is failing to render, with this error:

<img width="935" alt="image" src="https://github.com/elastic/kibana/assets/7405507/9387f1ff-e18f-4732-83db-4db2633ea75a">